lms/djangoapps/branding/api.py

```diff
@@ -329,3 +329,63 @@ def _absolute_url_staticfile(is_secure, name):
     # For local development, the returned URL will be relative,
     # so we need to make it absolute.
     return _absolute_url(is_secure, url_path)
+
+
+def get_microsite_url(name):
+    """
+    Look up and return the value for given url name in microsite configuration.
+    Return '#' if given url name is not defined in microsite configuration.
+    """
+    url = microsite.get_value(name)
+    if url:
+        return url
+    else:
+        return EMPTY_URL
+
+
+def get_url(name):
+    """
+    Lookup and return page url, lookup is preformed in the following order
+
+    1. get marketing link, if marketing is off then platform url will be used
+    2. override page url with microsite url if present
+
+    :return: string containing page url.
+    """
```

Comment:
I'd prefer to see a more concise implementation here, as long as it still results in the proper behavior -- for example, something along the lines of:

```
# If a microsite URL override exists, return it.  Otherwise return the marketing URL.
url = get_microsite_url(name)
if url != EMPTY_URL:
    return url
return marketing_link(name)
```

Also, JFTR, I really don't like the fact that we've introduced the concept of the edX marketing site into the Open edX codebase.  I don't have a better solution at the moment (override_url?) but something to think about.